### ISSUE TITLE
[🐛 Bug]: isPromise() returning false for thenable objects

### ISSUE THREAD

**Issue by cstringer** [SeleniumHQ/selenium#11045]
### What happened?

https://github.com/SeleniumHQ/selenium/commit/84dd6109ce692788467432ccee55f481fe49f2bc#r84520555

In the above commit, the `isPromise()` utility function was simplified to look for the return from `Object.prototype.toString()` to be`'[object Promise]'`, however this fails the function's JSDoc documentation that any object exposing a `then()` method would be considered a Promise (for example a `WebElementPromise` returns `'[object Object]'`).

My scripts that rely on `WebDriver.wait(<WebElementPromise>)` currently fail, but work after reverting the above change.

I propose that the previous functionality is returned to the `isPromise()` function to allow thenable objects to be treated as Promises.

Note: this may relate to the issue https://github.com/SeleniumHQ/selenium/issues/9348

### How can we reproduce the issue?

```
Pass an object with a `then()` method (for example a `WebElementPromise`) to `isPromise()`, it will return `false`.
```

### Relevant log output

```shell
n/a
```

### Operating System

Mac OS

### Selenium version

selenium-webdriver@4.4.0

### What are the browser(s) and version(s) where you see this issue?

Chrome

### What are the browser driver(s) and version(s) where you see this issue?

ChromeDriver 105.0.5195.19 (b9c217c128c16f53d12f9a02933fcfdec1bf49af-refs/branch-heads/5195@{#176})

**Comment by cstringer** [comment]
so sorry for that first PR, was moving a bit too fast this morning 🤦

**Comment by harsha509** [comment]
Thank you for the PR @cstringer !

Closing this as resolved as PR #11048 is landed !
